### PR TITLE
upstream `echo-server` with alb group and logging

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Echo server upstream
  * ALB Group support
  * ALB Logging support

## why
* Upstream
* Better test of all our featuresets

### CODE REVIEW

File: modules/eks/echo-server/main.tf [hunk 1]
```diff
@@ -50,8 +59,25 @@ module "echo_server" {
...
+      value = var.alb_access_logs_s3_bucket_name == null ? "" : var.alb_access_logs_s3_bucket_name
+      type  = "auto"
+    },
+    {
+      name  = "ingress.alb.access_logs.s3_bucket_prefix"
+      value = var.alb_access_logs_s3_bucket_prefix
+      type  = "auto"
     }
-  ]
+    ],
```
Comment (nitrocode) [inline modules/eks/echo-server/main.tf:78]: The formatting seems a little off here. did `terraform fmt` run ?

File: modules/eks/echo-server/remote-state.tf [hunk 2]
```diff
@@ -6,3 +6,13 @@ module "eks" {
 
   context = module.this.context
 }
+
+module "alb-controller-ingress-group" {
```
Comment (nitrocode) [inline modules/eks/echo-server/remote-state.tf:10]: Dashes shouldnt be used in resource or module names. Let's change to underscore.

```suggestion
module "alb_controller_ingress_group" {
```

File: modules/eks/echo-server/variables.tf [hunk 3]
```diff
@@ -6,5 +6,35 @@ variable "region" {
...
+  default     = "eks/cluster"
+}
+
+variable "alb_controller_ingress_group_component_name" {
+  type        = string
+  description = "The name of the alb-controller-ingress-group component"
+  default     = "eks/alb-controller-ingress-group"
+}
+
+variable "enable_alb_controller_ingress_group" {
```
Comment (nitrocode) [inline modules/eks/echo-server/variables.tf:18]:
```suggestion
variable "alb_controller_ingress_group_enabled" {
```

File: modules/eks/echo-server/main.tf [hunk 4]
```diff
@@ -2,6 +2,15 @@ locals {
   enabled               = module.this.enabled
   ingress_nginx_enabled = var.ingress_type == "nginx" ? true : false
   ingress_alb_enabled   = var.ingress_type == "alb" ? true : false
+
+  alb_access_logs_enabled = var.alb_access_logs_enabled && var.alb_access_logs_s3_bucket_name != null && var.alb_access_logs_s3_bucket_name != ""
+  ingress_controller_group_enabled = var.enable_alb_controller_ingress_group ? [
```
Comment (nitrocode) [inline modules/eks/echo-server/main.tf:7]:
```suggestion
  ingress_controller_group_enabled = var.alb_controller_ingress_group_enabled ? [
```

File: modules/eks/echo-server/main.tf [hunk 5]
```diff
@@ -2,6 +2,15 @@ locals {
...
+    {
+      name  = "ingress.alb.group_name"
+      value = module.alb-controller-ingress-group.outputs.group_name
```
Comment (nitrocode) [inline modules/eks/echo-server/main.tf:10]:
```suggestion
      value = module.alb_controller_ingress_group.outputs.group_name
```

File: modules/eks/echo-server/main.tf (hunk 1 above)
Comment (Benbentwo) [reply modules/eks/echo-server/main.tf:78]: Yeah I ran it that's how it formats it...

**Review by nitrocode — DISMISSED** [review]

**Review by nitrocode — APPROVED** [review]

**Review by nitrocode — CHANGES_REQUESTED** [review]
please fix pre-commit before merging

**Review by nitrocode — APPROVED** [review]

**Pull request merged**